lib/ldclient-rb/ldclient.rb

```diff
@@ -39,30 +39,35 @@ def initialize(api_key, config = Config.default)
       @worker = create_worker()
     end
 
+    def flush()
```

Comment:
Good catch. Changed so that we handle the case where the `pop` method has fewer elements than we expect. The underlying queue is thread-safe, so the rest of this should be ok.